lib/Regex/RegexParser.cpp

```diff
@@ -828,6 +850,29 @@ class Parser {
             return ClassAtom(CharacterClass::Words, ec == 'W' /* invert */);
           }
 
+#ifdef HERMES_ENABLE_UNICODE_REGEXP_PROPERTY_ESCAPES
+          case 'p':
+          case 'P': {
+            if (flags_.unicode) {
+              consume(ec);
+              std::string propertyName;
+              std::string propertyValue;
+              if (!tryConsume('{') ||
+                  !tryConsumeUnicodePropertyValueExpression(
```

Comment:
Yes, your description seems like what I had in mind. The `ClassAtom` would just store an `ArrayRef` to the data that we later need to add to the bracket, which is more in line with how it is used today.

> Which validation are you referring to? That the property name + property value are a known Unicode property?

Yes, exactly.

> creating a std::vector

I think that an `ArrayRef` would be sufficient. The idea is to just persist the `rangeArrayPool` variable currently in the `addUnicodePropertyRanges` function through the `ClassAtom`. That splits the function so that the validation happens first, and the mechanical task of adding the ranges can happen when we populate the bracket.